store/src/main/java/org/apache/rocketmq/store/schedule/ScheduleMessageService.java

```diff
@@ -260,6 +260,16 @@ private long correctDeliverTimestamp(final long now, final long deliverTimestamp
             return result;
         }
 
+        private void scheduleNextDeliverTask(long nextOffset, long countdown) {
+            if (ScheduleMessageService.this.isStarted()) {
```

Comment:
DeliverDelayedMessageTimerTask has been judged here, I want to know what is the basis for adding judgment here。

Reply:
thanks for reply. please check issue #3229 which provide some stack trace log.

if one thread call shutdown ScheduleMessageService. the timer will be canceled.

and at the same time when the timer running in `executeOnTimeup` method. (which may cost a long time)
the latter thread may also schedule a new DeliverDelayedMessageTimerTask to run another round.
but the current code not check the timer status. and we got an IllegalStateException here.